api/internal/handler/config_migrate/migrate.go

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package migrate
+
+import (
+	"encoding/binary"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+
+	"github.com/apisix/manager-api/internal/handler"
+	"github.com/apisix/manager-api/internal/utils/consts"
+	"github.com/gin-gonic/gin"
+	"github.com/shiningrush/droplet/data"
+
+	"hash/crc32"
+
+	"github.com/apisix/manager-api/internal/core/migrate"
+	"github.com/apisix/manager-api/internal/log"
+)
+
+const (
+	exportFileName = "apisix-config.bak"
+)
+
+type Handler struct{}
+
+func NewHandler() (handler.RouteRegister, error) {
+	return &Handler{}, nil
+}
+
+func (h *Handler) ApplyRoute(r *gin.Engine) {
+	r.GET("/apisix/admin/migrate/export", h.ExportConfig)
+	r.POST("/apisix/admin/migrate/import", h.ImportConfig)
+}
+
+type ExportInput struct{}
+
+func (h *Handler) ExportConfig(c *gin.Context) {
+	data, err := migrate.Export(c)
+	if err != nil {
+		log.Errorf("Export: %s", err)
+		c.JSON(http.StatusInternalServerError, err)
+		return
+	}
+	// To check file integrity
+	// Add 4 byte(uint32) checksum at the end of file.
+	checksumUint32 := crc32.ChecksumIEEE(data)
+	checksum := make([]byte, 4)
+	binary.BigEndian.PutUint32(checksum, checksumUint32)
+	fileBytes := append(data, checksum...)
+
+	c.Writer.WriteHeader(http.StatusOK)
+	c.Header("Content-Disposition", "attachment; filename="+exportFileName)
+	c.Header("Content-Type", "application/octet-stream")
+	c.Header("Accept-Length", strconv.Itoa(len(fileBytes)))
+	c.Header("Content-Transfer-Encoding", "binary")
+	_, err = c.Writer.Write([]byte(fileBytes))
+	if err != nil {
+		log.Errorf("Write: %s", err)
+	}
```

Comment:
Need to give an error response.

Reply:
This is a err sending data to client.If we can't "Write",we can't give an error response.